### PR TITLE
Fix body_json parsing of api_request module

### LINKED ISSUE #334: [Help]: linode.cloud.api_request
### Description

I'm trying to issue a task which would reboot my vm into rescue mode, but I'm getting error suggesting my JSON is incorrect. I'm pretty sure JSON is not missing double-quotes anywhere, so I must be doing something wrong and can't see where.

### Ansible Configuration

CONFIG_FILE() = /opt/ansible/ansible.cfg
DEFAULT_VAULT_PASSWORD_FILE(/opt/ansible/ansible.cfg) = /opt/ansible/.vault-pass
HOST_KEY_CHECKING(/opt/ansible/ansible.cfg) = False

### Attempted Config

Below fails with `fatal: [localhost]: FAILED! => {"changed": false, "msg": "JSONDecodeError: Expecting property name enclosed in double quotes: line 1 column 2 (char 1)"}`:
```yml
# empty_linode registered earlier
- name: Boot empty linode into rescue mode
  # https://github.com/linode/ansible_linode/blob/dev/docs/modules/api_request.md
  linode.cloud.api_request:
    # https://www.linode.com/docs/api/linode-instances/#linode-boot-into-rescue-mode
    api_token: "{{ linode_api_token }}"
    path: "/linode/instances/{{ empty_linode.instance.id }}/rescue"
    method: POST
    body_json: >
      {
        "devices": {
          "sda": {
            "disk_id": {{ empty_linode.configs[0].devices.sda.disk_id }}
          }
        }
      }
  register: response
```

However below "works" (linode boots to rescue mode however no devices are mapped):
```yml
# empty_linode registered earlier
- name: Boot empty linode into rescue mode
  # https://github.com/linode/ansible_linode/blob/dev/docs/modules/api_request.md
  linode.cloud.api_request:
    # https://www.linode.com/docs/api/linode-instances/#linode-boot-into-rescue-mode
    api_token: "{{ linode_api_token }}"
    path: "/linode/instances/{{ empty_linode.instance.id }}/rescue"
    method: POST
    body_json: >
      {
        "devices": {
        }
      }
  register: response
```

I'm trying to use json above because below yml renders to json with strings as disk id (int is required), it would be nice if in this case strings could be converted to int when rendering to json
```yml
- name: Boot empty linode instance into rescue mode
  # https://github.com/linode/ansible_linode/blob/dev/docs/modules/api_request.md
  linode.cloud.api_request:
    # https://www.linode.com/docs/api/linode-instances/#linode-boot-into-rescue-mode
    api_token: "{{ linode_api_token }}"
    path: "/linode/instances/{{ empty_linode.instance.id }}/rescue"
    method: POST
    body:
      devices:
        sda:
          disk_id: "{{ empty_linode.configs.devices.sda.disk_id }}"
  register: response
```

### PR DESCRIPTION
**Pull request by zliang-akamai**
## 📝 Description

closes #334

Translate `body_json` parameter to json to fix an issue that trouble the users.

## ✔️ How to Test

`make TEST_ARGS="-v api_request_extra" test`
`make TEST_ARGS="-v api_request_basic" test`

### CODE REVIEW

**Comment by lgarber-akamai** [comment]
/acctest tests=api_request_extra sha=9838372270f5a8af29a759bfbe4687065f488f6b

**Review by lgarber-akamai — CHANGES_REQUESTED** [review]
Great work! This solution is really clever.
Just a small refactor and we should be good to go 👍

File: plugins/modules/api_request.py [hunk 1]
```diff
@@ -133,6 +136,12 @@ def exec_module(self, **kwargs: Any) -> Optional[dict]:
         param_body = self.module.params.get("body")
         param_body_json = self.module.params.get("body_json")
 
+        with contextlib.suppress(Exception):
+            parsed_body_json = ast.literal_eval(param_body_json)
+            parsed_body_json = json.dumps(parsed_body_json)
+
+            param_body_json = parsed_body_json
+
```
[the remark this answers is not in the document]
Comment (zliang-akamai) [reply plugins/modules/api_request.py]: Applied. Thank you for the suggestion!

**Review by lgarber-akamai — APPROVED** [review]
Nice work!

**Comment by zliang-akamai** [comment]
/acctest tests=api_request_extra sha=62158be127c60fdd86de588abcdf1caf060b4297

**Comment by zliang-akamai** [comment]
/acctest tests=api_request_extra sha=62158be127c60fdd86de588abcdf1caf060b4297

**Comment by zliang-akamai** [comment]
/acctest tests=api_request_extra sha=c9beefb17ea551a72ce0975da0466acadf5652bc

**Comment by zliang-akamai** [comment]
/acctest tests=api_request_extra sha=c9beefb17ea551a72ce0975da0466acadf5652bc

**Comment by zliang-akamai** [comment]
/acctest tests=api_request_extra sha=0560d90f86ff96e44a0919c504dfa0bcfa4ba8a2

**Comment by zliang-akamai** [comment]
/acctest tests=api_request_extra sha=cb460c0495e52ec6d0ebfec3f1281d9e9506e44e

**Comment by zliang-akamai** [comment]
/acctest tests=api_request_extra sha=cb460c0495e52ec6d0ebfec3f1281d9e9506e44e

**Review by yec-akamai — APPROVED** [review]

**Pull request merged**